digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/strategy/DefaultSectionCalculatorStrategy.scala

```diff
@@ -123,33 +148,31 @@ class DefaultSectionCalculatorStrategy extends RoadAddressSectionCalculatorStrat
       link.flatMap(pl => GeometryUtils.calculatePointFromLinearReference(pl.geometry, calibrationPoint.segmentMValue))
     }
 
-    //First it will return the subtraction between endPoint - startPoint of all the points variable
-    //Then it will sum all the results of the previous operation as a vector (v1 is the accumulated result, v2 is the next item to sum
-    def calculateDirectionVector(projectLinkSeq: Seq[ProjectLink]): Vector3d = {
-      projectLinkSeq.map(pl => (pl.startingPoint, pl.endPoint)).map(p => p._2 - p._1).fold(Vector3d(0, 0, 0)) { case (v1, v2) => v1 + v2 }.normalize2D()
-    }
-
     // Pick the one with calibration point set to zero: or any old link with lowest address: or new links by direction
     calibrationPoints.find(_.addressMValue == 0).flatMap(calibrationPointToPoint).getOrElse(
       oldLinks.filter(_.status == LinkStatus.UnChanged).sortBy(_.startAddrMValue).headOption.map(_.startingPoint).getOrElse {
         val remainLinks = oldLinks ++ newLinks
         if (remainLinks.isEmpty)
           throw new InvalidAddressDataException("Missing right track starting project links")
-        //Grab all the endpoints of the links
-        val points = remainLinks.map(pl => (pl.startingPoint, pl.endPoint))
-        //TODO: ORIGINAL Direction calculation
-        val direction = points.map(p => p._2 - p._1).fold(Vector3d(0, 0, 0)) { case (v1, v2) => v1 + v2 }.normalize2D()
-        /*TODO: VIITE-1601 direction calculation
-        val direction = remainLinks.exists(_.sideCode != SideCode.Unknown) match {
-          case true =>
-            //We use the points that already have a side code defined as the basis to find the direction vector
-            calculateDirectionVector(remainLinks.filter(_.sideCode != SideCode.Unknown))
-          case _ =>
-            calculateDirectionVector(remainLinks)
+        // Grab all the endpoints of the links
+        val direction = if (remainLinks.exists(_.sideCode != SideCode.Unknown)) {
+          remainLinks.filter(_.sideCode != SideCode.Unknown).map(p => p.endPoint - p.startingPoint).fold(Vector3d(0, 0, 0)) { case (v1, v2) => v1 + v2 }.normalize2D()
+        } else {
+          val sampleLink = remainLinks.head.geometry.head
+          if (remainLinks.flatMap(_.geometry).forall(geo => geo.y == sampleLink.y)) {
+            // Very rare case
+            // ALL Y Values are equal on the tracks check the X Value
```

Comment:
Does this mean that y-values at starting and ending points are the same?

Reply:
@skosonen Exactly, the very rare case where the whole collective of project links represent a perfectly parallel line to the X axis. In that case there is no variation on the Y values.

Reply:
Line doesn't have to be straight, just if the starting and ending points have the same y-values, then the addressing should grow from west to the east (left to the right).

Reply:
@skosonen All right, doing that change then.

Reply:
This will not work when there is at least one project link with different Y values. Because then the direction vector used for the project link with the same Y value will be also (0.0, 1.0, 0.0)